scheduler/plugin/wasm_test.go

```diff
@@ -0,0 +1,177 @@
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package wasm
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"sigs.k8s.io/kube-scheduler-wasm-extension/scheduler/test"
+)
+
+var wasmMagicNumber = []byte{0x00, 0x61, 0x73, 0x6d}
+
+func Test_getURL(t *testing.T) {
+	testCtx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	bin, err := os.ReadFile(test.URLTestAllNoopWat[7:])
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, _ = w.Write(bin)
+	}))
+	t.Cleanup(ts.Close)
+
+	type testCase struct {
+		name          string
+		url           string
+		expected      []byte
+		expectedError string
+	}
+
+	tests := []testCase{
+		{
+			name:     "file valid",
+			url:      test.URLTestAllNoopWat,
+			expected: bin,
+		},
+		{
+			name:          "empty url",
+			expectedError: "missing url",
+		},
+		{
+			name:          "http invalid",
+			url:           "http:// ",
+			expectedError: `parse "http:// ": invalid character " " in host name`,
+		},
+		{
+			name:          "https invalid",
+			url:           "https:// ",
+			expectedError: `parse "https:// ": invalid character " " in host name`,
+		},
+		{
+			name:     "http",
+			url:      ts.URL,
+			expected: bin,
+		},
+
+		{
+			name:          "unsupported scheme",
+			url:           "ldap://foo/bar.wasm",
+			expectedError: "unsupported URL scheme: ldap",
+		},
+		{
+			name:          "file not found",
+			url:           "file://testduta",
+			expectedError: "open testduta: ",
+		},
+		{
+			name: "file dir not file",
+			url:  "file://.",
+			// Below ends in "is a directory" in unix, and "The handle is invalid." in windows.
+			expectedError: "read .: ",
+		},
+	}
+
+	for _, tt := range tests {
+		tc := tt
+		t.Run(tc.name, func(t *testing.T) {
+			bin, err := getURL(testCtx, tc.url)
+			if tc.expectedError != "" {
+				// Use substring match as the error can be different in Windows.
+				if !strings.Contains(err.Error(), tc.expectedError) {
+					t.Fatalf("expected err %v to contain %s", err, tc.expectedError)
+				}
+			} else if err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+			if want, have := tc.expected, bin; !bytes.Equal(want, have) {
+				t.Fatalf("unexpected binary: want %v, have %v", want, have)
+			}
+		})
+	}
+}
+
+func Test_httpGet(t *testing.T) {
+	wasmBinary := wasmMagicNumber
+	wasmBinary = append(wasmBinary, 0x00, 0x00, 0x00, 0x00)
```

Comment:
ps I have no idea why we are appending 4 bytes as this was copied from before.. possibly to give some more data to compress?

Reply:
I think it was padded to make it a valid wasmBinary, but this could be misguided. I didn't think of verifying without the padding.